src/libZilliqa/Zilliqa.cpp

```diff
@@ -274,10 +274,12 @@ Zilliqa::Zilliqa(const PairOfKey& key, const Peer& peer, SyncType syncType,
         break;
       case SyncType::RECOVERY_ALL_SYNC:
         LOG_GENERAL(INFO, "Recovery all nodes");
+        if (m_mediator.m_lookup->GetSyncType() == SyncType::RECOVERY_ALL_SYNC) {
```

Comment:
I think this check will never go true, because we never call SetSyncType(SyncType::RECOVERY_ALL_SYNC)